doc/WebSite/Entity-History.md

```diff
@@ -113,9 +113,13 @@ While you can select tracked entities by configuration, you can use the
 ```
 
 All properties of MyEntity are tracked except MyProperty2 since it's
-explicitly disabled. The Audited attribute can be used to
+explicitly disabled. The **Audited** attribute can be used to
 save change logs for a desired property.
 
+When using **Audited** attribute on **entity** class, **EntityChange will always be created** for entity in *Addeded/Modified/Deleted* state, regardless of whether any **PropertyChange** is being created.
+
+Similarly, when using **Audited** attribute on a **property**, **PropertyChange will always be created** for entity in *Addeded/Modified/Deleted* state, regardless of whether there is any differences between the **new and old values** of the **property**.
```

Comment:
As mentioned in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5126#discussion_r357987933, this behaviour is neither consistent nor correct.

This (second) phrasing appears even more forced and does not make any sense:
- `Audited` attribute on **entity** class → `EntityChange` will be created based on **entity** state. (Correct)
- `Audited` attribute on **property** → `PropertyChange` will be created based on **entity** state — what? (Incorrect)

Instead, we should fix the regression from #4870.

Reply:
> Audited attribute on property → PropertyChange will be created based on entity state — what? (Incorrect)

@acjh can you explain in details?  Since `EntityChange` is only created with entity has changes, and for `PropertyChange` to be created, `EntityChange` need to be created before hand right?

Reply:
A property change results in an entity change, not the other way around.

A `PropertyChange` itself does not need an `EntityChange` to be created, but is logically accessed only via `EntityChange.PropertyChanges`.

- `EntityChange` should only be created if **entity** is changed.
- `PropertyChange` should only be created if **property** is changed.

`PropertyChange` should **not** be created just because entity has (other) changes.

Reply:
> **PropertyChange will be created**, regardless of whether there is any differences between the **new and old values** of the **property**.

Still not right. *`PropertyChange` should only be created if property is changed.*